lib/mongo/grid/fs_bucket.rb

```diff
@@ -487,8 +487,28 @@ def files_name
 
       def ensure_indexes!
         if files_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
-          chunks_collection.indexes.create_one(FSBucket::CHUNKS_INDEX, :unique => true)
-          files_collection.indexes.create_one(FSBucket::FILES_INDEX)
+          unless index_exists?(files_name, FSBucket::FILES_INDEX)
+            files_collection.indexes.create_one(FSBucket::FILES_INDEX)
+          end
+        end
+
+        if chunks_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
+          unless index_exists?(chunks_name, FSBucket::CHUNKS_INDEX)
+            chunks_collection.indexes.create_one(FSBucket::CHUNKS_INDEX, :unique => true)
+          end
+        end
+      end
+
+      # Helper method to test for index existence by first verifying the collection exists.
+      # This allows calling Mongo::Collection#indexes without raising
+      #   Mongo::Error::OperationFailure Exception: [26:NamespaceNotFound]
+      # if the collection doesn't exist
```

Comment:
I'll note that the "Enumerating indexes" spec [contains a suggested implementation](https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst#enumeration-getting-index-information) on how to enumerate indexes. That implementation checks for the `NamespaceNotFound` error and returns an empty array in that case.

If the driver should continue raising an exception when listing indexes on non-existing collections, this method should catch the exception itself and return `false` in that case. This also removes the need for running two commands for an index check.

Reply:
+1

Reply:
Rewrote the logic here